rotkehlchen/chain/ethereum/modules/aave/decoder.py

```diff
@@ -0,0 +1,127 @@
+                event.location_label == to_address and
+                event.balance.amount == amount
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Stake {amount} AAVE'
+                if from_address != to_address:
+                    event.notes += f' for {to_address}'
+                event.product = EvmProduct.STAKING
+                event.counterparty = CPT_AAVE
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset.identifier == STKAAVE_IDENTIFIER and
+                event.location_label == to_address and
+                event.balance.amount == amount
+            ):
+                event.counterparty = CPT_AAVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {event.balance.amount} stkAAVE from staking in Aave'
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_unstake(
+            self,
+            from_address: ChecksumEvmAddress,
+            to_address: ChecksumEvmAddress,
+            amount: FVal,
+            decoded_events: list[EvmEvent],
+    ) -> DecodingOutput:
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset.identifier == STKAAVE_IDENTIFIER and
+                event.location_label == to_address and
+                event.balance.amount == amount
+            ):
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.notes = f'Unstake {amount} stkAAVE'
+                if from_address != to_address:
+                    event.notes += f' for {to_address}'
+                event.product = EvmProduct.STAKING
+                event.counterparty = CPT_AAVE
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_AAVE and
+                event.location_label == to_address and
+                event.balance.amount == amount
+            ):
+                event.counterparty = CPT_AAVE
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Receive {event.balance.amount} AAVE after unstaking from Aave'
+
+        return DEFAULT_DECODING_OUTPUT
+
+    # DecoderInterface method
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        decoders = super().addresses_to_decoders()
+        decoders.update({STK_AAVE_ADDR: (self._decode_staking_events,)})
+        return decoders
```

Comment:
Moving this meant no unnecessary work here

```suggestion
        return {STK_AAVE_ADDR: (self._decode_staking_events,)}
```